python/ray/rllib/common.py

```diff
@@ -30,16 +59,22 @@ class Algorithm(object):
   TODO(ekl): support checkpoint / restore of training state.
   """
 
-  def __init__(self, env_name, config):
+  def __init__(self, env_name, config, s3_bucket=None):
+    self.experiment_id = uuid.uuid4()
```

Comment:
I'm a bit confused about how `s3_bucket` is being used. It looks like we're dumping the config info to it, but it looks like it isn't being used for anything else, is that correct? Wouldn't we want to at least save the bucket or the open handle to the s3 bucket in a field of the object? Or is the plan to do that later?